pkg/lib/indexer/indexer.go

```diff
@@ -291,25 +306,27 @@ func (i ImageIndexer) PruneFromIndex(request PruneFromIndexRequest) error {
 	return nil
 }
 
-// ExtractDatabase sets a temp directory for unpacking an image
-func (i ImageIndexer) ExtractDatabase(buildDir, fromIndex, caFile string, skipTLS bool) (string, error) {
+// ExtractDatabases sets a temp directory for unpacking an image's sqllite database and package configs folder
+func (i ImageIndexer) ExtractDatabases(buildDir, fromIndex, caFile string, skipTLS bool) (string, string, error) {
```

Comment:
I don't think we want to couple database extraction with config extraction. I feel it would be cleaner for each to have a dedicated method for extraction (or for there to be an interface for deserialization that we can switch between)

At a higher level, we should probably be treating these formats as equivalents (i.e. `sqlite=config`); defining a common intermediate representation for interacting with data, manipulate it, then convert to the target from our IR directly.

Reply:
Isn't `extract contents of the index` one task? I'd argue that the database and the config are one as `contents` of the index, not separate concerns. And that becomes apparent when I think about writing two routines for extracting them: the routines are almost identical, expect for what file path they're targeting inside the container.

Reply:
>  Isn't extract contents of the index one task?

The tasks share the same goal, but may achieve them differently -- I think you're picking up on the need for an abstraction over the extraction of content from different types of catalogs to the filesystem.

Consider what happens to this method signature when we deprecate sqlite indexes; removal of a return value will require us to update the logic in all code that depends on it. Similarly, if we ever add a new type of index, we'd potentially add a new return value, also requiring an update. Abstracting over a shared method signature gives us the opportunity to reduce the number of places we need to make changes (think dependency injection) and increases the opportunities for code reuse.